StringPermutation.java

```diff
@@ -0,0 +1,102 @@
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Generates all permutations of the given string using a backtracking
+ * algorithm.In this program , we start fixing letters from start to the end of
+ * the string.
+ * 
+ * @author Udit Saxena
+ * 
+ */
+public class StringPermutation {
+
+	List<String> permutedList = new ArrayList<String>();
```

Comment:
Instead of creating this variable in class, this can be used in generatePermutations method and then passed for recursion.